packages/app/src/runner/event-manager.ts

```diff
@@ -451,7 +451,7 @@ export class EventManager {
             Cypress.runner.resumeAtTest(runState.currentId, runState.emissions)
           }
 
-          return run()
+          return this.ws.emit('before:driver:run', run)
```

Comment:
Not sure if it totally matters, but do we want to only emit this if the protocol is enabled? We know that via `Cypress.config('protocolEnabled')`

Reply:
Good idea. Addressed in https://github.com/cypress-io/cypress/pull/27508/commits/139ac715d7a3ad2adcafc48ab294439f9d0fe744